### PR TITLE
Major Refactoring of the code

### PR DESCRIPTION
**Pull request by profxj**
What started off as handling `healpix` and `wcs` in a 
better manner became a major refactoring.

Am hoping that the `PATH` class will make it easier 
for non-experts to use the code.

### CODE REVIEW

File: docs/localization.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+the ellipse, the semi-major axis ["a", in arcsec], 
+the semi-minor axis ["b", in arcsec], and the PA on the sky (
+"theta", defined in wacky astronomer fashion, i.e. deg E of N).
+
+Here is an example::
+
+    frb_coord = SkyCoord('21h44m25.255s -40d54m00.10s', frame='icrs')
+    eellipse = dict(a=0.1, b=0.1, theta=0.)
+    localiz = dict(type='eellipse', center_coord=frb_coord, eellipse=eellipse)
+    assert localization.vette_localization(localiz)
```
Comment (KshitijAggarwal) [inline docs/localization.rst:28]: typo. It should be vet_localization

File: astropath/path.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        self.raw_prior_Oi = priors.raw_prior_Oi(
+            self.cand_prior['P_O_method'], self.candidates['ang_size'], 
+            mag=self.candidates['mag'] if 'mag' in self.candidates.keys() else None)
+
+        # Normalize
+        self.prior_Oi = priors.renorm_priors(self.raw_prior_Oi, 
+                                             self.cand_prior['P_U'])
+
+        # Add to candidate table
+        self.candidates['P_O'] = self.prior_Oi
```
Comment (KshitijAggarwal) [inline astropath/path.py:141]: Would be useful to add logging/print statements here like: 
"Calculating priors"
"Rernormalizing priors"
"Adding prior values to candidates dataframe"

**Review by KshitijAggarwal — COMMENTED** [review]
Looks good!

**Comment by KshitijAggarwal** [comment]
Two more things: 
1. It would be better to use [`logging`](https://docs.python.org/3/howto/logging.html) instead of prints. 
2. It might be cleaner to use different classes for `Candidates`, `Localisation` and `Priors` instead of dmodel and vetting structure.

File: docs/localization.rst (hunk 1 above)
Comment (profxj) [reply docs/localization.rst:28]: fixed

File: astropath/path.py (hunk 2 above)
Comment (profxj) [reply astropath/path.py:141]: done

**Review by KshitijAggarwal — APPROVED** [review]

**Review by KshitijAggarwal — APPROVED** [review]

**Pull request merged**